### PR TITLE
feat: support the new SSO config format

### PR DESCRIPTION
**Pull request by anaisberg**
- solves this issue: [New SSO config format](https://github.com/thomasmichaelwallace/serverless-better-credentials/issues/18)

### CODE REVIEW

**Comment by ealain** [comment]
Thank you for this work!

Could you create a commit with only the feature part, without linting? This would make the review easier, particularly the test part.

I'd recommend creating additional commits, if you think linting is required.

File: src/types.ts [hunk 1]
```diff
@@ -44,8 +45,22 @@ export type ClientInfo = {
...
-  sso_region: string,
-  sso_role_name: string,
-  sso_start_url: string,
+  sso_account_id: string;
+  sso_region: string;
+  sso_role_name: string;
+  sso_start_url: string;
+  sso_session?: string;
+  region?: string;
+  output?: string;
```
Comment (guiyom-e) [inline src/types.ts]: no need to add these two new keys
```suggestion
```

**Comment by gozineb** [comment]
Thank you so much for this fix @anaisberg , been struggling with this !

File: src/types.ts [hunk 2]
```diff
@@ -48,4 +49,16 @@ export type SsoProfileConfig = {
...
+
+export type ConfigData = {
+  config: IniFileContent;
+  keys: string[];
+  values: Record<string, string>[];
+};
+
+export type SsoIniLoader =
+   IniLoader & {
+     loadSsoSessionsFrom(options: LoadFileOptions): IniFileContent;
```
Comment (guiyom-e) [inline src/types.ts:63]: It won't be necessary anymore when https://github.com/aws/aws-sdk-js/pull/4456 is merged.

Comment (anaisberg) [reply src/types.ts:63]: that's great!!

**Comment by bneigher** [comment]
oh yes - looking forward to this being merged!!

**Comment by guiyom-e** [comment]
@thomasmichaelwallace any plan for a new release ?

**Comment by ThomasAribart** [comment]
Awesome 🤩

**Comment by thomasmichaelwallace** [comment]
OK - let me see about getting this made!

**Comment by thomasmichaelwallace** [comment]
checkout version 1.2.0 - feel free to open a PR if it doesn't act like you expect.

(I rarely use this project these days, I made the leap to aws-cdk and haven't really looked back!)

**Pull request merged**